cmake/functions.cmake

```diff
@@ -194,12 +194,14 @@ function(ml_add_test_executable _target)
         ${CMAKE_CURRENT_BINARY_DIR}/$<IF:$<CONFIG:Release>,Release,Debug>/ml_test_${_target}.exe
         ${CMAKE_CURRENT_BINARY_DIR}/ml_test_${_target}.exe
       COMMAND ${CMAKE_CURRENT_BINARY_DIR}/ml_test_${_target}
+      COMMENT "Running test: ml_test_${_target}"
       WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
       )
   else()
     add_custom_target(test_${_target}
       DEPENDS ml_test_${_target}
-      COMMAND ${CMAKE_CURRENT_BINARY_DIR}/ml_test_${_target}
+      COMMAND ${CMAKE_CURRENT_BINARY_DIR}/ml_test_${_target} 2>&1 | tee ${CMAKE_CURRENT_BINARY_DIR}/ml_test_${_target}.log
```

Comment:
I think it's best not to use `tee`, as this will still cause the CI job output to be unreadable, with output from multiple processes interleaved.

What would be nice is to do something similar to Gradle, where it hides the output in a file _unless_ there's a failure, in which case it prints it all at the end.

So we'd need to do something like:

1. Run `ml_test` and redirect the output to a file
2. Remember the exit code of `ml_test`
3. If the exit code is not zero, `cat` the file
4. Make the success of the whole step depend not on the exit code of `cat` but the one we remembered from `ml_test`

Hopefully by doing a full `cat` at the end there'll be less interleaving than there is if `ml_test` writes to `stderr` as it runs.